src/auth/AuthController.ts

```diff
@@ -2,13 +2,27 @@ import { AuthError } from '../types/api'
 import { NextFunction, Request, RequestHandler, Response } from 'express'
 
 class AuthController {
-    getAccountId = function (authToken: string): number {
-        // dirty hack to integrate new podcast engineering kiosk
-        // should be kind of a database later
-        if (authToken === 'Bearer engkiosk-81ztQL36e') {
-            return 2
+    accountsMap: { [key: string]: number }
+
+    constructor(accountsMap: { [key: string]: number }) {
+        this.accountsMap = accountsMap
+    }
+
+    /**
+     * Returns the account id for the specified auth token
+     * @param authToken The auth token to get the account id for (incl. 'Bearer ')
+     * @returns The account id
+     * @throws AuthError if the token is not valid
+     * */
+    getAccountId = (authToken: string): number => {
+        // remove 'Bearer ' from token to get the actual token
+        const token = authToken.substring(7)
+
+        if (this.accountsMap[token] !== undefined) {
+            return this.accountsMap[token]
+        } else {
+            throw new AuthError('Specified token is not valid')
```

Comment:
I find this a little more readable. It also checks the `Bearer` prefix so that we indeed remove the correct part of the token and have at least a length of 7.

```suggestion
    getAccountId = (authToken: string): number => {
        if (!authToken.startsWith('Bearer ')) {
            throw new AuthError('Specified token is not valid')
        }
    
        // Remove 'Bearer ' from token to get the actual token
        const token = authToken.substring(7)
    
        const accountId = this.accountsMap[token]
        if (accountId === undefined) {
            throw new AuthError('Specified token is not valid')
        }
        return accountId
```

Reply:
this check starts with is already outside of the function https://github.com/openpodcast/api/blob/89677bbdf2c2699d7e3f387ff816514b4c84bd0d/src/auth/AuthController.ts#L46 but we can also add it here again

Reply:
Guess I'd add (or move) it here. Then it's easier to write unit tests for that part as the logic is in one place.

Reply:
added in 89677bbdf2c2699d7e3f387ff816514b4c84bd0d